tests/src/Interop/MarshalAPI/NativeLibrary/NativeLibraryTests.cs

```diff
@@ -0,0 +1,396 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using TestLibrary;
+
+using Console = Internal.Console;
+
+enum TestResult {
+    Success,
+    ReturnFailure,
+    ReturnNull,
+    IncorrectEvaluation,
+    ArgumentNull,
+    ArgumentBad,
+
+    DllNotFound,
+    BadImage,
+    InvalidOperation,
+    EntryPointNotFound,
+    GenericException
+    };
+
+public class NativeLibraryTest
+{
+    static string CurrentTest;
+    static bool Verbose = false;
+ 
+    public static int Main()
```

Comment:
Please add negative tests that ensure we don't load libraries from places that we shouldn't. E.g. that we don't load from the application or working directory if given a full path and that we're correctly piping DllImportSearchPath flags